client.go

```diff
@@ -79,12 +79,21 @@ type federationTripper struct {
 	transportsMutex sync.Mutex
 	skipVerify      bool
 	resolutionCache sync.Map // serverName -> []ResolutionResult
+	dnsCache        *DNSCache
 }
 
-func newFederationTripper(skipVerify bool) *federationTripper {
+func newFederationTripper(skipVerify bool, options ...interface{}) *federationTripper {
+	var dnsCache *DNSCache
+	for _, opt := range options {
+		switch o := opt.(type) {
+		case *DNSCache:
+			dnsCache = o
```

Comment:
Generally the idiom is to use `WithDNSCache(thing)` instead of this. Can we make it use that shape instead?

Reply:
It is done with a `WithDNSCache` struct.